### ISSUE TITLE
Switch from token-based to OIDC-ish login

### ISSUE THREAD

**Issue by nyiyui** [wlmac/metropolis#249]
Currently, the app and React-based client uses the `/api/auth/token` endpoint with a username+password pair to login the user.
`/api/auth/token` relies on JWTs (pros/cons listed below), and 

Pros to `/api/auth/token`
- nothing that I can think of
Cons to `/api/auth/token`:
- extra RSA certificate to take care of
- non-instant token expiry (might be wrong, but I'm not aware of such a feature in the Django backend code)
- more attack surface (3 ways of logging in - `/api/auth/token`, Django (web+admin), OAuth/OIDC-ish)

Two ways to move forward:
- app and client uses a authentication proxy server (authproxy) (have to check if this is secure) with Resource-Owner Password (ROP) flow
  - authproxy required to make OAuth client confidential
- app and client migrates to OIDC (no ROP flow)

**Comment by JasonLovesDoggo** [comment]
no authproxy. we do **not** need an extra service

**Comment by nyiyui** [comment]
Agreed... I would've loved writing security-critical code in RISC assembly, though! (just kidding)

**Comment by JasonLovesDoggo** [comment]
> Agreed... I would've loved writing security-critical code in RISC assembly, though! (just kidding)

I don't trust your "secure" code.... you've printed passwords & allowed anyone to bypass all authentication on api 😨

**Comment by nyiyui** [comment]
That's true! I'm sure I've learned lessons from them, and 90% sure I won't make those mistakes again (by staying away from security code...)!

**Comment by JasonLovesDoggo** [comment]
> That's true! I'm sure I've learned lessons from them, and 90% sure I won't make those mistakes again (by staying away from security code...)!

to anyone reading this in the future... ken is a very trustworthy and cool person!!! HIRE THEM (and me too pls I'm probably struggling and jobless.....wow look at my cool contribution history)
